src/test/scala/com/microsoft/sqlserver/jdbc/spark/bulkwrite/DataSourceTest.scala

```diff
@@ -142,4 +154,16 @@ class DataSourceTest extends FunSuite with Matchers {
             "dataPoolDataSource"->"myds"))
         assert(true == DataPoolUtils.isDataPoolScenario(options))
     }
+    
+    test("Correct AAD driver is set when accessToken is specified") {
```

Comment:
This sets the options.  Would be good to validate in the test that actually got set as expected. Also test for other options added encrypt, hostnameCertificate. Also having a defaut value test is important.

Reply:
Added a test for default values and also added the above mentioned option to the existing test.